server/src/test/java/org/opensearch/cluster/routing/allocation/NodeVersionAllocationDeciderTests.java

```diff
@@ -671,6 +668,98 @@ public void testRebalanceDoesNotAllocatePrimaryOnHigherVersionNodesSegrepEnabled
         );
     }
 
+    public void testCanAllocatePrimaryOnHigherVersionNodesDocRepEnabled() {
```

Comment:
Since SETTING_REPLICATION_TYPE setting defaults to document replication, so could you check if this new test case is duplicated with the existing test cases?